### ISSUE TITLE
Improvements for distinguishing between card colors for a color blind person

### ISSUE THREAD

**Issue by MidasLamb** [ParabolInc/parabol#5776]
One of my colleagues is color blind and has trouble distinguishing the color of the cards when in the retro meeting in the discussion phase. 

Currently the color palette when creating a template has 12 colors. Which makes it hard to pick four colors our colleague can easily distinguish. 

I see some alternative solutions which could all help with this problem, ordered by (I think) amount of effort required to implement it:
* Allow a free choice of RGB value (potentially in an "advanced menu")
* Allow colored icons/patterns rather then only a solid color
* Allow a special "color blind" mode per user where an individual user can map the preexisting 12 colors to colors suitable for them (for when more than 1 color blind person is in a team)

**Comment by jordanh** [comment]
@MidasLamb we're reviewing this issue in our backlog refinement meeting now. Thanks _a ton_ for opening this issue

**Comment by jordanh** [comment]
## Enhancement

### Acceptance Criteria
   - A set of low-fidelity designs exploring the ideas above
   - A summary artifact with a recommendation for which one to pursue, including a linked backlog issue articulating the next step
   - Feedback collected from @MidasLamb & colleague

### Estimated Effort
   - 13 hours

**Comment by jordanh** [comment]
@MidasLamb may we connect with you for design feedback when we have some things to share?

**Comment by MidasLamb** [comment]
@jordanh , certainly! 
I'm not sure what the exact timeline is here, but the holidays start soon so I won't be able to relay it back to the colorblind colleague until after the holidays

**Comment by enriquesanchez** [comment]
Hi @MidasLamb 👋

Big thank you for submitting this issue and your help.

Prompted by your suggestions, @acressall and I have been looking at solutions to make the category of the reflection cards less dependent on color alone as a means to be identifiable. 

We started exploring an alternative that adds a text label (thus not relying on color to be identifiable) and would love to hear you and your colleague's thoughts on it.

## What we are proposing

As it currently stands, a reflection's category is communicated via the color circle on the top-left corner of a card:

<img width="515" alt="Captura de Pantalla 2022-01-28 a la(s) 3 27 25 p m" src="https://user-images.githubusercontent.com/3276087/151623802-59a383e0-833e-4c19-9f2b-95740f81d2dc.png">

Because the surface area is a bit small, we found that adding a texture or pattern on top does not work that well. So instead we started to explore the text label idea:


<img width="516" alt="Captura de Pantalla 2022-01-28 a la(s) 3 22 31 p m" src="https://user-images.githubusercontent.com/3276087/151623276-8eb26b2c-aef0-4457-ac56-5a654fc22641.png">

We find that the text label adds important context to the reflection, making it easier to identify not only for color blind users, but also to anyone using the app.

The color would still be used, but it would no longer be the only means to identify the type of reflection.

<img width="1076" alt="Captura de Pantalla 2022-01-28 a la(s) 3 28 45 p m" src="https://user-images.githubusercontent.com/3276087/151623982-4319b253-0b27-4af8-8be5-f70e0544aed5.png">

Again, thank you so much for these suggestions and we'd love to hear what you and your colleague think. Also pinging @ackernaut (Parabol's design lead) for input and feedback.


## Some additional details

We figure some users might prefer the simplicity of the current style, so we could have these expand/collapse and remember that choice as a user preference:

https://user-images.githubusercontent.com/3276087/151627536-c7c531e2-2b05-4221-95d7-be2facd4e57a.mp4

And while most common prompts will fit nicely, there's always a chance for a longer prompt to exist. In that scenario we think a little truncation can help:

<img width="589" alt="Captura de Pantalla 2022-01-28 a la(s) 4 03 48 p m" src="https://user-images.githubusercontent.com/3276087/151627989-7853a6d9-43b5-4ec5-a288-964856ed0b7d.png">

**Comment by MidasLamb** [comment]
@enriquesanchez These look great IMO!! 🎉 
 I'll check with my colleague when I can coming week! If you haven't heard back from by end of next week, feel free to ping me again here, sometimes I can get quite forgetful!

**Comment by MidasLamb** [comment]
@enriquesanchez , my colleague also said it would work great for them!

**Comment by mattkrick** [comment]
My .02:

- one challenge with this is card height. looks like this'll increase height by ~32px, which means a user will likely see 2+ fewer cards per column before scrolling. For posterity, this is also the reason we don't support reactjis (during grouping) or non-anonymous reflections-- we want to be hyper focused on the card content to cram as much as we can in the viewport!
- I'm always a fan of iconography (and color) over text. It takes less space & a user can comprehend an icon faster than they can some text-- this is especially true if the language isn't my native one! That's huge for being inclusive of cross-cultural teams.

to support our colorblind audience, i'd propose we combine the effort with our work on dark mode. we can offer 4 color palettes: light, dark, colorblind light, colorblind dark. i've had some tension that we  this is how github does it

![Screen Shot 2022-02-08 at 9 52 21 AM](https://user-images.githubusercontent.com/5514175/153047689-48b2417a-3590-4725-883f-3403888e779a.png)

**Comment by ackernaut** [comment]
@enriquesanchez can you link to the Figma files for your exploration?

Here are my thoughts:
- I’ve often treated the prompt as secondary information. My assumption is that folks only look for the prompt occasionally. I’m probably wrong. However, when they want to know it, it needs to be easier
- Today folks can hover to read the prompt, the tooltip having decent readability, but I’m not sure folks always discover that
- This increases the overall height of the cards, of course, and may push a few more out of view in a column, but we have spotlight now so if I really have trouble finding a card I can use search. And I’d probably have the same trouble finding what I want regardless of the label height
- We could consider tucking the text label in the corner. [This example](https://www.figma.com/file/sp3B3eQjrGKBS2tayJmlwY/Reflection-Card?node-id=101%3A73) takes a 2-line card from 64 px to 80 px tall. It’s a little funky (the pills you used have a cleaner rhythm) and just ‘a different idea’ that’s easier to visual quickly that try to imagine:
![image](https://user-images.githubusercontent.com/307286/153272888-22391ad8-6ea6-431e-851c-72315ad1fef5.png)
- Re: the animated collapsing concept, I assume that if I collapse the label, the user preference sticks like that for me, and happens to all the cards? I like the idea in general. One note: a lot of layout is calculated on the height of these cards, for example, the stack layout, the grid layout, grouping animations. So if we make it where the label can collapse and expand, dynamically changing the height, we’d have to have the conditional size throughout all of those layouts and animations, and it would have to look good. If I collapse my view, what will happen to animations are in flight? It’ll probably be fine. It’s not undoable but not just as simple.
- We need to define a text color per palette color (as your concept shows). This mapping could probably be reusable elsewhere when we use tags. To get close to best a11y colors, we’d have to assign white for the bolder colors and a dark text color for the lighter palette colors. I think if we use a 14px 600 font style we’re probably good enough with the right text color
- Overall I think your concepts are safe to try. Thank you for giving this some love

**Comment by jordanh** [comment]
Personally I like the idea of making this a theme, like with what @mattkrick said

**Comment by ackernaut** [comment]
@enriquesanchez pushing these designs forward is available
